### ISSUE TITLE
Unexpected behaviour using front matter key "url" alongside "uglyURLs = true" *for list pages

### ISSUE THREAD

**Issue by AesaraB** [gohugoio/hugo#10393]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.104.3+extended linux/amd64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?
This is the latest release.

### Issue
When setting a url in front matter, such as `url = '/section/test'`, the output file is `/section/test/index.html`. However, when configuration `uglyURLs = true` is set, the output file remains `/section/test/index.html`, instead of `/section/test.html` which is consistent with other ugly url output files.

This behaviour may cause issues with certain web server configurations.

Update: This issue only occurs for list pages.

**Comment by bep** [comment]
What happens if you do `url = '/section/test.html'`?

To me the `url` attribute is a way to be explicit about the URL to use which would also be a way to override any global url setting (e.g. uglyURLs ...).

**Comment by AesaraB** [comment]
Hi, sorry for not putting this into the original post -- I've already tried this and unfortunately it does not work.

**Comment by AesaraB** [comment]
Wait apologies, it fails in a different way. Let me look into this a bit more

**Comment by AesaraB** [comment]
Now this is interesting, on single pages appending the `url` value with `.html` appears to work, but on list pages it renders an RSS template.

**Comment by jmooring** [comment]
@AesaraB 

> but on list pages it renders an RSS template

I am unable to reproduce this behavior.

site configuration

```toml
uglyURLs = true
disableKinds = ['RSS','sitemap','taxonomy','term']  # Disabled to simplify rendered site
```

content directory

```text
content/
├── posts/
│   ├── _index.md
│   ├── post-1.md
│   └── post-2.md
└── _index.md
```

content/posts/_index.md

```text
+++
title = 'Posts'
date = 2022-11-03T18:56:29-07:00
draft = false
url = '/posts/test.html'
+++
```

rendered site

```
public/
├── posts/
│   ├── post-1.html
│   ├── post-2.html
│   └── test.html
└── index.html
```

I think this is working as designed, and the issue should be closed.

**Comment by AesaraB** [comment]
@jmooring

I've put ``disabledKinds = ['RSS']`` in configuration and set the url frontmatter for a list page to end in .html and this issue is no long present. Can you remove RSS from the disabledKinds and try this again? Thanks

**Comment by jmooring** [comment]
> Can you remove RSS from the disabledKinds and try this again? 

Sure. The result:

```text
public/
├── posts/
│   ├── post-1.html
│   ├── post-2.html
│   └── test.html
├── index.html
└── index.xml
```

Try it:

```bash
git clone --single-branch -b hugo-github-issue-10393 https://github.com/jmooring/hugo-testing hugo-github-issue-10393
cd hugo-github-issue-10393
rm -rf public/ && hugo && tree public
```

**Comment by AesaraB** [comment]
If that works for you then there may be an issue with my build. See the output html files here [disabledKinds.zip](https://github.com/gohugoio/hugo/files/9934637/disabledKinds.zip)

**Comment by AesaraB** [comment]
@jmooring

Without disabling RSS, can you view the contents ``public/posts/test.html`` in a browser?

**Comment by jmooring** [comment]
> Without disabling RSS, can you view the contents of public/posts/test.html in a browser 

Yes. Why don't you try the example site from my previous comment?

This isssue is closed. If you need additional assistance, please use the forum (https://discourse.gohugo.io/) for questions and troubleshooting. We prefer to use GitHub for verified bugs and vetted enhancements. Thanks.

**Comment by jmooring** [comment]
OK, I'm reopening because it fails intermittently.

```bash
hugo --printPathWarnings
```

> WARN 2022/11/03 20:42:13 Duplicate target paths: /posts/test.html (2)
